be/src/exprs/cast_functions.cpp

```diff
@@ -229,6 +229,17 @@ StringVal CastFunctions::cast_to_string_val(FunctionContext* ctx, const StringVa
   return sv;
 }
 
+BooleanVal CastFunctions::cast_to_boolean_val(FunctionContext* ctx, const StringVal& val) {
+    if (val.is_null) return BooleanVal::null(); 
+        StringParser::ParseResult result;
+        BooleanVal ret;
+        ret.val = StringParser::string_to_bool(reinterpret_cast<char*>(val.ptr), val.len, &result);
```

Comment:
StringParser::string_to_bool will consider "1" to false, is it ok?

Reply:
"1" will convert to int  not bool